pyogrio/tests/conftest.py

```diff
@@ -18,14 +18,35 @@ def pytest_report_header(config):
     )
 
 
+def prepare_testfile(testfile_path, dst_dir, ext):
+    if ext == testfile_path.suffix:
+        return testfile_path
+    else:
+        dst_path = dst_dir / f"{testfile_path.stem}{ext}"
+        if dst_path.exists():
+            return dst_path
+        gdf = pyogrio.read_dataframe(testfile_path)
+        pyogrio.write_dataframe(gdf, dst_path)
+        return dst_path
+
+
 @pytest.fixture(scope="session")
 def data_dir():
     return _data_dir
 
 
-@pytest.fixture(scope="session")
-def naturalearth_lowres():
-    return _data_dir / Path("naturalearth_lowres/naturalearth_lowres.shp")
+@pytest.fixture(scope="function")
+def naturalearth_lowres(tmp_path, request):
+    ext = request.param \
+            if ("param" in dir(request) and request.param is not None) else ".shp"
+    testfile_path = _data_dir / Path("naturalearth_lowres/naturalearth_lowres.shp")
+
+    return prepare_testfile(testfile_path, tmp_path, ext)
+
+
+@pytest.fixture(scope="function", params=[".shp", ".gpkg", ".json"])
```

Comment:
Agreed with other comments that `.` can be dropped from inputs and handled within paths inside the fixture.